### ISSUE TITLE
Not enough value in generating FASTA files

### ISSUE THREAD

**Issue by tdw1221** [griffithlab/pVACtools#553]
I ran into the following error when trying to use PVACSEQ predicting for .vep.vcf file:

```
Generating Variant Peptide FASTA and Key Files - Entries 18801-19000
Traceback (most recent call last):
  File "/ysm-gpfs/home/dt485/anaconda2/bin/pvacseq", line 10, in <module>
    sys.exit(main())
  File "/ysm-gpfs/home/dt485/anaconda2/lib/python3.6/site-packages/tools/pvacseq/main.py", line 99, in main
    args[0].func.main(args[1])
  File "/ysm-gpfs/home/dt485/anaconda2/lib/python3.6/site-packages/tools/pvacseq/run.py", line 172, in main
    pipeline.execute()
  File "/ysm-gpfs/home/dt485/anaconda2/lib/python3.6/site-packages/lib/pipeline.py", line 468, in execute
    self.generate_fasta(chunks)
  File "/ysm-gpfs/home/dt485/anaconda2/lib/python3.6/site-packages/lib/pipeline.py", line 282, in generate_fasta
    fasta_generator.execute()
  File "/ysm-gpfs/home/dt485/anaconda2/lib/python3.6/site-packages/lib/fasta_generator.py", line 160, in execute
    wildtype_amino_acid, mutant_amino_acid = line['amino_acid_change'].split('/')
ValueError: not enough values to unpack (expected 2, got 1)
```

It looks like there is a missing WT or MT amino acid information, and it should be somewhere between 18801-19000 entries. I wonder is there a way I could pin down to the specific mutation according to the entries? I'm predicting for length = 8,9,10,11

**Comment by susannasiebert** [comment]
There should be a `<sample_name>.tsv_<chunk>` file in your output folder. I think for these entries the chunk would be 9401-9500. In there, check the `amino_acid_change` column and find the entry or entries that don't have a `/` in it. If you could post the variant here so I can investigate further, that would be great.

**Comment by tdw1221** [comment]
> There should be a `<sample_name>.tsv_<chunk>` file in your output folder. I think for these entries the chunk would be 9401-9500. In there, check the `amino_acid_change` column and find the entry or entries that don't have a `/` in it. If you could post the variant here so I can investigate further, that would be great.

Thanks Susanna, I've managed to find it by looking into pipeline.py. It was because the annotated amino acid change looks like: G/-, I deleted that one and it worked. 

Just another quick question, it seems that the output filtered neoantigen list tend to limit to one neoantigen per missense variant, I wonder how exactly is the criterion is since I'm manually filtering neoantigen from all epitope file, thanks!

**Comment by susannasiebert** [comment]
Huh, I wouldn't expect this variant to be causing that sort of error though since the amino acid change field does contain a `/` (i.e., `line['amino_acid_change'].split('/')` would be able to split such a value into two parts). Do you mind sharing a VCF with just this variant with us so we can debug further?

In regards to the filtered file, one of the filters applied is the top score filter which will only keep the best-binding epitope for each variant. If different transcripts of a variant result in different mutant peptide sequence (e.g., due to differing splice sites) then this filter will return the top epitope for each transcript that are non-identical. The logic is roughly like this:
- Find the best-binding epitope for each transcript
- Group transcripts by their best-binding epitope since some transcripts might result in the same mutant peptide sequence and other transcripts might result in a differing mutant peptide sequence so some groups of transcripts might each share the same best-binding epitope.
- For each group above, if multiple transcripts result in the same best-binding epitope, keep only the transcript for this epitope with the highest transcript expression. If no transcript expression information is available, return the transcript-epitope combination for the transcript with the lowest Ensembl identifier.
For more information see https://pvactools.readthedocs.io/en/latest/pvacseq/filter_commands.html#top-score-filter.
It's important to note that this filter runs after the binding, coverage, and tsl filters. If you want to manually filter your all_epitopes.tsv file you may want to consider using our standalone filtering command described on the page I linked to above. They will use the same logic as the pipelines but can play around with the different thresholds.

**Comment by tdw1221** [comment]
Thank you Susanna, regarding the missing amino acid change, I also deleted one with only -, not / at all. then I guess that's the problem. Thank you again for this clear clarification, very much appreciate it!

**Comment by susannasiebert** [comment]
I would still be interested to see the variant that was causing this problem (the one with `-`). This is something pVACseq should handle so you don't have to manually delete any variant like it. If you'd be willing to share a VCF with just the variant with use, that would be greatly appreciated.

**Comment by tdw1221** [comment]
Sure, the whole file is around 181MB, what is the best way to give it to
you?

Susanna Kiwala <notifications@github.com> 于2020年5月13日周三 下午12:25写道：

**Comment by susannasiebert** [comment]
Can you upload it to Google Drive and share it with ssiebert.gc@gmail.com?

**Comment by tdw1221** [comment]
just did, feel free to let me know how it goes, thank you!

Susanna Kiwala <notifications@github.com> 于2020年5月13日周三 下午12:31写道：

**Comment by susannasiebert** [comment]
I'm reopening this issue so I don't forget to this debug this case.

**Comment by susannasiebert** [comment]
This bug has been fixed in version 1.5.9.
